StationeersLaunchPad/LoadedMod.cs

```diff
@@ -27,49 +27,86 @@ public class LoadedMod
     public ContentHandler ContentHandler;
 
     public List<Type> SMEntryTypes = new();
-    public List<Type> BepinexEntryTypes = new();
+    public List<Type> BepInExEntryTypes = new();
     public List<(Type, MethodInfo)> DefaultEntryTypes = new();
     public List<GameObject> EntryPrefabs = new();
 
     public List<ConfigFile> ConfigFiles = new();
 
-    public bool LoadFinished = false;
-    public bool LoadFailed = false;
+    public bool LoadedAssemblies;
+    public bool ResolvedAssemblies;
+    public bool LoadedAssets;
+    public bool LoadedEntryPoints;
+    public bool LoadFinished;
+    public bool LoadFailed;
 
     public LoadedMod(ModInfo info)
     {
-      this.Logger = Logger.Global.WithPrefix($"[{info.DisplayName}]");
+      this.Logger = Logger.Global.WithPrefix($"[{info.DisplayName}]: ");
       this.Info = info;
       var resource = new DummyResource(info.Path);
       this.ContentHandler = new(resource, new List<IResource>().AsReadOnly(), this.Prefabs.AsReadOnly());
     }
 
+    private async UniTask<Assembly> LoadAssemblySingle(AssemblyInfo assemblyInfo)
+    {
+      this.Logger.LogDebug($"Loading Assembly {assemblyInfo.Name}");
+      var assembly = await ModLoader.LoadAssembly(assemblyInfo.Path);
+      ModLoader.RegisterAssembly(assembly, this);
+      this.Logger.LogDebug($"Loaded Assembly");
+      return assembly;
+    }
+
     public async UniTask LoadAssembliesSerial()
     {
       foreach (var assemblyInfo in this.Info.Assemblies)
-        this.Assemblies.Add(await this.LoadAssembly(assemblyInfo));
+        this.Assemblies.Add(await this.LoadAssemblySingle(assemblyInfo));
     }
+
     public async UniTask LoadAssembliesParallel()
     {
       var assemblies = await UniTask.WhenAll(
-        this.Info.Assemblies.Select(assemblyInfo => this.LoadAssembly(assemblyInfo))
+        this.Info.Assemblies.Select(assemblyInfo => this.LoadAssemblySingle(assemblyInfo))
       );
       this.Assemblies.AddRange(assemblies);
     }
 
-    public void ResolveAssemblies()
+
+    private async UniTask ResolveAssembliesSingle(Assembly assembly)
+    {
+      assembly.GetTypes();
```

Comment:
this is cpu-bound so for it to actually run in parallel gotta wrap it in `UniTask.RunOnThreadPool`